cmd/soroban-rpc/internal/methods/simulate_transaction.go

```diff
@@ -21,19 +21,21 @@ type SimulateTransactionCost struct {
 	MemoryBytes     uint64 `json:"memBytes,string"`
 }
 
-type SimulateTransactionResult struct {
-	Auth      []string `json:"auth"`
-	Events    []string `json:"events"`
-	Footprint string   `json:"footprint"`
-	// TODO: update documentation and review field name
-	XDRs []string `json:"xdrs"`
+// SimulateHostFunctionResult contains the simulation result of each HostFunction within each InvokeHostFunctionOp
+type SimulateHostFunctionResult struct {
+	Auth []string `json:"auth"`
+	XDR  string   `json:"xdr"`
```

Comment:
I think this may need to include `hostFunctionIndex` and `operationIndex` to provide association, looking at it from upstream usage, it will need to match up each sim result instance here to a InvokeHostFunctionOp::HostFunction instance in the user's tx so that it can set the `auth` from sim result. or I may be mis-understanding intent of usage.

Reply:
oh, I think understand further, the association back to source/user's tx can be derived from the serialized order of these in `SimulateTransactionResponse.Results`, the first `SimulateHostFunctionResult` belonging to first InvokeHostFunctionOp and first HostFunction within source/user's tx and so on.

Reply:
There can only be a single InvokeHostFunctionOp, so each result refers to each HostFunction

Reply:
got it, thanks, I read into the docs stated on line 24 too literally, and thought the single-op restriction had been lifted, just for consideration, could state that a bit more directly   `SimulateHostFunctionResult contains the simulation result of each HostFunction within the single InvokeHostFunctionOp allowed on a Tx`

Reply:
Good point. I've updated the comment